### PR TITLE
User/bsmithvt/extension grant

### PR DESCRIPTION
**Pull request by bsmithvt**
This change is [Reviewable](https://reviewable.io/reviews/healthcatalyst/fabric.identity/155)

### CODE REVIEW

File: Fabric.Identity.API/Models/CustomGrantType.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Fabric.Identity.API.Models
+{
+    public static class CustomGrantType
```
Comment (michaelvidal24) [inline Fabric.Identity.API/Models/CustomGrantType.cs]: Is this an API model, or is it more of a constant? If its a constant, should we move it to the constants namespace?

File: Fabric.Identity.API/Validation/DelegationGrantValidator.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using IdentityModel;
+using IdentityServer4.Models;
+using IdentityServer4.Validation;
+
+namespace Fabric.Identity.API.Validation
```
Comment (michaelvidal24) [inline Fabric.Identity.API/Validation/DelegationGrantValidator.cs]: Should we consider moving this to a new namespace called Fabric.Identity.API.ExtensionGrantValidators? While it is named *Validator, its not really synonymous with our other validators that validate API models.

File: Fabric.Identity.API/Models/CustomGrantType.cs (hunk 1 above)
Comment (bsmithvt) [reply Fabric.Identity.API/Models/CustomGrantType.cs]: Good point - IS4 put their GrantType.cs in a Models namespace, but I agree our Models namespaces has a different meaning. I'll move it.

File: Fabric.Identity.API/Validation/DelegationGrantValidator.cs (hunk 2 above)
Comment (bsmithvt) [reply Fabric.Identity.API/Validation/DelegationGrantValidator.cs]: Yes, I agree. Will move this as well.

**Pull request merged**